documentation/dotnet-monitor.md

```diff
@@ -0,0 +1,276 @@
+
+```bash
+docker pull mcr.microsoft.com/dotnet/nightly/monitor:5.0.0-preview1
+```
+
+### Running `dotnet-monitor`
+
+Once installed you can run `dotnet-monitor` via the following command,
+
+```bash
+dotnet monitor collect
+```
+
+When running in a docker container, you will need to share a volume mount between your application container and `dotnet-monitor`
+
+```bash
+docker volume create diagnosticserver
+docker run -d --rm -p 8000:80 -v diagnosticsserver:/tmp mcr.microsoft.com/dotnet/core/samples:aspnetapp
+docker run -it --rm -p 52323:52323 -v diagnosticsserver:/tmp mcr.microsoft.com/dotnet/nightly/monitor --urls http://*:52323
+```
+
+When running in cluster, it is recommend to run the `dotnet-monitor` container as a sidecar alongside your application container in the pod. This sample Kubernetes manifest shows how to configure your deployment to include a sidecar container.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: dotnet-hello-world
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: dotnet-hello-world
+  template:
+    metadata:
+      labels:
+        app: dotnet-hello-world
+    spec:
+      volumes:
+      - name: diagnostics
+        emptyDir: {}
+      containers:
+      - name: server
+        image: mcr.microsoft.com/dotnet/core/samples:aspnetapp
+        ports:
+        - containerPort: 80
+        volumeMounts:
+          - mountPath: /tmp
+            name: diagnostics
+      - name: sidecar
+        image: mcr.microsoft.com/dotnet/nightly/monitor
+        ports:
+        - containerPort: 52325
+        # args: ["--urls", "http://*:52323", "--metricUrls", "http://*:52325"]
+        volumeMounts:
+          - name: diagnostics
+            mountPath: /tmp
+```
+
+
+
+In the default configuration `dotnet-monitor` binds to two different  groups of URLs. The URLs controlled via the `--urls` parameter (defaults to http://localhost:52323) expose all the collection endpoints. The URLs controlled via the ``--metricUrls` parameter (defaults to http://localhost:52325) only exposes the `/metrics` endpoint. Since diagnostics artifacts such as logs, dumps, and traces can leak sensitive information about the application, it is **strongly recommended** that you only expose the metrics URL in the container scenario.
+
+### Endpoints
+
+The REST API exposed by dotnet-monitor exposes the following endpoints:
+
+- `/processes`
+- `/dump/{pid?}`
+- `/gcdump/{pid?}`
+- `/trace/{pid?}`
+- `/logs/{pid?}`
+- `/metrics`
+
+In the sections that follow we'll look at the functionality exposed by each of these endpoints and how you can use these endpoints to collect diagnostics artifacts.
+
+### Processes
+
+The `/processes` endpoint returns a list of target processes accessible to `dotnet-monitor`. 
+
+> NOTE: When running locally the `dotnet-monitor` tools lists itself as one of the target processes. 
```

Comment:
I felt the need to call this out specifically since when dotnet-monitor detects only one target process, there isn't a need to explicitly specify a process id. Given that dotnet-monitor always detects itself, it breaks a big convenience in the tool and is worth calling out.

I will move it to the end of the section.

Reply:
And yes, you are right in that this is behavior we don't want.
https://github.com/dotnet/diagnostics/issues/1270